### DISCUSSION TITLE
How can I access GlueSQL virtual tables as SQLite tables via Rusqlite vtab interface?

### DISCUSSION THREAD

**Discussion by shah** [gluesql/gluesql#1442] [Q&A]
I have an existing SQLite database using Rusqlite and I'd like to use GlueSQL to access many external data sources as SQL tables. Instead of developing a new storage infrastructure, I'd like SQLite to store the data using something like:

```sql
create table sqlite_X as select * from glue_sql_Y
```

If I can make a GlueSQL table available through Rusqlite's vtab infrastructure, then the above should work. Has anyone done something similar?

**Comment by panarch** [comment]
No there does not exist solution for handling SQLite using GlueSQL yet.
there was a kind of plan to support GlueSQL SQL syntax to SQLite like SQL DBs but it will take some time.

making custom storage impl for SQLite would be more simple. then at least we can import data and manipulate it from SQLite easily through GlueSQL.
